source/slang/slang-legalize-types.h

```diff
@@ -357,38 +357,29 @@ struct SimpleLegalVarChain
     IRVarLayout* varLayout = nullptr;
 };
 
-/// A "chain" of variable declarations that can handle both primary and "pending" data.
+/// A "chain" of variable declarations
 ///
 /// In the presence of interface-type fields, a single variable may
 /// have data that sits in two distinct allocations, and may have
 /// `VarLayout`s that represent offseting into each of those
 /// allocations.
 ///
-/// A `LegalVarChain` tracks two distinct `SimpleVarChain`s: one for
-/// the primary/ordinary data allocation, and one for any pending
-/// data.
+/// A `LegalVarChain` tracks the primary/ordinary data allocation.
 ///
-/// It is okay if the primary/pending chains have different numbers
-/// of links in them.
-///
-/// Offsets for particular resource kinds in the primary or pending
-/// data allocation can be queried on the appropriate sub-chain.
+/// Offsets for particular resource kinds in the primary data
+/// allocation can be queried on the appropriate sub-chain.
 ///
 struct LegalVarChain
 {
     // The chain of variables that represents the primary allocation.
     SimpleLegalVarChain* primaryChain = nullptr;
```

Comment:
As mentioned in an earlier comment, the hope is that we can simply rename `SimpleLegalVarChain` to `LegalVarChain` and get rid of this wrapper type entirely. This wrapper didn't exist at all until the whole "pending layout" idea was introduced, and the code was at least marginally easier to follow without it.

(I know that the type legalization pass is one of the hardest parts of the codebase to follow already, so that isn't saying much...)